src/types/index.d.ts

```diff
@@ -58,3 +59,30 @@ type Keypair = {
 }
 
 type PackageMode = 'staging' | 'production' | 'development' | 'test'
+
+type VerificationOptions = {
+  secretKey?: string;
+  transactionExpiry?: number
+}
+
+type VerificationBasestringOptions = {
+  transactionId: string; 
+  formId: string; 
+  fieldId: string; 
+  answer: string; 
+  time: number 
+}
+
+type VerificationSignatureOptions = {
+  transactionId: string; 
+  formId: string; 
+  fieldId: string; 
+  answer: string;
+}
+
+type VerificationAuthenticateOptions = { 
+  signatureString: string; 
+  submissionCreatedAt: number; 
+  fieldId: string;
+  answer: string 
+}
```

Comment:
Types and Interfaces are quite similar in Typescript, but by composing the types it conveys its semantic meaning much more clearly:

```suggestion
// A verified answer contains a field ID and answer
type VerifiedAnswer = {
  fieldId: string; 
  answer: string;
}

// Add the transaction ID and form ID to a VerifiedAnswer to obtain a signature
type VerificationSignatureOptions = VerifiedAnswer & {
  transactionId: string; 
  formId: string; 
}

// Creating a basestring requires the epoch in addition to signature requirements
type VerificationBasestringOptions = VerificationSignatureOptions & { time: number }

// Authenticate a VerifiedAnswer with a signatureString and epoch
type VerificationAuthenticateOptions = VerifiedAnswer & { 
  signatureString: string; 
  submissionCreatedAt: number; 
}
```